sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -881,7 +881,16 @@ class Analyzer(
             }
           case u @ UnresolvedFunction(funcId, children, isDistinct) =>
             withPosition(u) {
-              catalog.lookupFunction(funcId, children) match {
+              val func = try {
+                catalog.lookupFunction(funcId, children)
+              } catch {
+                // SPARK-16228 ExternalCatalog may recognize `double`-type only.
+                case _: Exception =>
+                  val newChildren = children.map(x =>
```

Comment:
Shouldn't this be solved in the ExternalCatalog? This seems way to implementation specific to put in the Analyzer.